app/services/locale_normalizer.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class LocaleNormalizer
+  # Normalizes locale string, matching the list of I18n.locales where possible
+  # @param locale [String,Symbol] the locale to normalize
+  # @return [String] the normalized locale
+  def self.normalize_to_i18n(locale)
+    return nil if locale.blank?
+    locale = locale.to_s.gsub("-", "_")
+
+    i18n_pairs.each { |downcased, value| return value if locale.downcase == downcased }
+
+    locale
+  end
+
+  def self.is_same?(locale1, locale2)
```

Comment:
This method could do with an RDoc explaining what it does, and why, since you need to be familiar enough with locale naming schemes to understand the purpose.